model/task_lifecycle_test.go

```diff
@@ -772,6 +772,42 @@ func TestUpdateBuildStatusForTaskReset(t *testing.T) {
 	assert.Equal(t, evergreen.VersionStarted, data.Status)
 }
 
+func TestUpdateVersionStatusForGithubChecks(t *testing.T) {
+	require.NoError(t, db.ClearCollections(task.Collection, build.Collection, VersionCollection, event.AllLogCollection))
+	b1 := build.Build{
+		Id:                "b1",
+		Status:            evergreen.BuildStarted,
+		Version:           "v1",
+		Activated:         true,
+		IsGithubCheck:     true,
+		GithubCheckStatus: evergreen.BuildSucceeded,
+	}
+
+	b2 := build.Build{
+		Id:        "b2",
+		Status:    evergreen.BuildFailed,
+		Version:   "v1",
+		Activated: true,
+	}
+
+	assert.NoError(t, b1.Insert())
+	assert.NoError(t, b2.Insert())
+	v1 := Version{
+		Id:     "v1",
+		Status: evergreen.VersionStarted,
+	}
+	assert.NoError(t, v1.Insert())
+	versionStatus, err := UpdateVersionStatus(&v1)
+	assert.NoError(t, err)
```

Comment:
Should this `assert.NoError` (and the other `NoError` checks) be `require` since the test probably can't continue anyways if it fails to insert or `UpdateVersionStatus` fails?

Reply:
I typically prefer only using require for things that would then result in a nil pointer dereference.